config.go

```diff
@@ -394,6 +416,38 @@ func (cfg *Config) unshadow() {
 	cfg.shadow = nil
 }
 
+func generateClusterSecret() ([]byte, error) {
+	encodedSecretLength := 64
+	secret := make([]byte, base64.StdEncoding.DecodedLen(encodedSecretLength))
```

Comment:
This is weird. The secret is 32 bytes. This is not something that you should infer from the length of the encoded base64 representation of the secret (which by the way, is 48). You are making a longer than 32 byte secret here.